### ISSUE TITLE
Unable to link images

### ISSUE THREAD

**Issue by srid** [jgm/commonmark-hs#44]
The parser doesn't create correct pandoc AST nodes for this syntax:

```markdown
[![asciicast](https://asciinema.org/a/329911.svg)](https://asciinema.org/a/329911)
```

Here's how one would expect it to render:

[![asciicast](https://asciinema.org/a/329911.svg)](https://asciinema.org/a/329911)

**Comment by srid** [comment]
Workaround:

```markdown
[![emacsDemo]](https://asciinema.org/a/329911)

[emacsDemo]: https://asciinema.org/a/329911.svg
```

**Comment by jgm** [comment]
What pandoc AST are you getting, and what pandoc AST do you expect?

**Comment by srid** [comment]
Oh strange. It works now. I guess it was because I had removed the `autolink` extension. So same as #46?

```
     [Link
        ("", [], [])
        [Image
           ("", [], [])
           [Str "asciicast"]
           ("https://asciinema.org/a/329911.svg", "")]
        ("https://asciinema.org/a/329911", "")]
```

**Comment by jgm** [comment]
I think that with the fix to #46 it works even with `autolinks`:
```
% commonmark -xall --json | pandoc -f json -t native
[![asciicast](https://asciinema.org/a/329911.svg)](https://asciinema.org/a/329911)
^D
[Para
  [Link ("",[],[])
    [Image ("",[],[])
    [Str "asciicast"]
    ("https://asciinema.org/a/329911.svg","")]
    ("https://asciinema.org/a/329911","")]]
```
